src/Controller/ResourceController.php

```diff
@@ -262,7 +262,7 @@ public function handle(Request $request) {
       $response->addCacheableDependency($render_context);
     }
     foreach ($parameters as $parameter) {
-      $response->addCacheableDependency($parameter);
+      $response->addCacheableDependency(is_array($parameter) ? reset($parameter) : $parameter);
```

Comment:
If `$parameter` is an array shouldn't we loop through each item? We should add cacheable dependency for all items, not just the first.